be/src/olap/page_cache.cpp

```diff
@@ -21,35 +21,81 @@ namespace doris {
+    }
+    else if (index_cache_ratio == 1) {
+        _index_page_cache = std::unique_ptr<Cache>(new_lru_cache("IndexPageCache", capacity));
+    }
+    else if (index_cache_ratio > 0 && index_cache_ratio < 1) {
+        _data_page_cache = std::unique_ptr<Cache>(new_lru_cache("DataPageCache", static_cast<size_t>(capacity * (1 - index_cache_ratio))));
+        _index_page_cache = std::unique_ptr<Cache>(new_lru_cache("IndexPageCache", static_cast<size_t>(capacity * index_cache_ratio)));
+    }   
+}
 
-bool StoragePageCache::lookup(const CacheKey& key, PageCacheHandle* handle) {
-    auto lru_handle = _cache->lookup(key.encode());
-    if (lru_handle == nullptr) {
+bool StoragePageCache::lookup(const CacheKey& key, PageCacheHandle* handle, segment_v2::PageTypePB page_type) {
+    Cache::Handle* lru_handle = nullptr;
+    switch (page_type) {
+    case segment_v2::DATA_PAGE:
+        lru_handle = _data_page_cache->lookup(key.encode());
+        if (lru_handle == nullptr) {
+            return false;
+        }
+        *handle = PageCacheHandle(_data_page_cache.get(), lru_handle);
+        break;
+    case segment_v2::INDEX_PAGE:
+        lru_handle = _index_page_cache->lookup(key.encode());
+        if (lru_handle == nullptr) {
+            return false;
+        }
+        *handle = PageCacheHandle(_index_page_cache.get(), lru_handle);
+        break;
+    default:
         return false;
     }
-    *handle = PageCacheHandle(_cache.get(), lru_handle);
     return true;
 }
 
 void StoragePageCache::insert(const CacheKey& key, const Slice& data, PageCacheHandle* handle,
-                              bool in_memory) {
+                              segment_v2::PageTypePB page_type, bool in_memory) {
     auto deleter = [](const doris::CacheKey& key, void* value) { delete[](uint8_t*) value; };
 
     CachePriority priority = CachePriority::NORMAL;
     if (in_memory) {
         priority = CachePriority::DURABLE;
     }
 
-    auto lru_handle = _cache->insert(key.encode(), data.data, data.size, deleter, priority);
-    *handle = PageCacheHandle(_cache.get(), lru_handle);
+    Cache::Handle* lru_handle = nullptr;
+    switch (page_type) {
+    case segment_v2::DATA_PAGE:
+        lru_handle = _data_page_cache->insert(key.encode(), data.data, data.size, deleter, priority);
+        *handle = PageCacheHandle(_data_page_cache.get(), lru_handle);
+        break;
+    case segment_v2::INDEX_PAGE:
+        lru_handle = _index_page_cache->insert(key.encode(), data.data, data.size, deleter, priority);
+        *handle = PageCacheHandle(_index_page_cache.get(), lru_handle);
+        break;
+    default:
+        break;
+    }
+}
+
+bool StoragePageCache::is_cache_available(segment_v2::PageTypePB page_type) {
+    if (page_type == segment_v2::DATA_PAGE && _index_cache_ratio == 1) {
+        return false;
+    }
+    else if (page_type == segment_v2::INDEX_PAGE && _index_cache_ratio == 0) {
+        return false;
+    }
```

Comment:
If you do the refactor as mentioned above, you can simply judge whether the return `lru_handle ` is nullptr or not